From:  Duncan Overbruck <mozilla@duncano.de>

Profiler will fail to build on musl other non GLIBC toolchains

https://bugzilla.mozilla.org/show_bug.cgi?id=1358214

diff --git a/tools/profiler/core/platform.h b/tools/profiler/core/platform.h
--- a/tools/profiler/core/platform.h
+++ b/tools/profiler/core/platform.h
@@ -39,17 +39,17 @@
 #include "mozilla/UniquePtr.h"
 #include "mozilla/Unused.h"
 #include "PlatformMacros.h"
 #include <vector>
 #include "StackTop.h"
 
 // We need a definition of gettid(), but glibc doesn't provide a
 // wrapper for it.
-#if defined(__GLIBC__)
+#if defined(__linux__) || !defined(__BIONIC__)
 #include <unistd.h>
 #include <sys/syscall.h>
 static inline pid_t gettid()
 {
   return (pid_t) syscall(SYS_gettid);
 }
 #elif defined(GP_OS_darwin)
 #include <unistd.h>
